controllers/flowcollector_controller.go

```diff
@@ -67,15 +84,41 @@ func (r *FlowCollectorReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		},
 		OperatorNamespace: operatorNamespace,
 	}
-	err = gfReconciler.Reconcile(ctx, &desired.Spec.GoflowKube)
-	if err != nil {
+	if err := gfReconciler.Reconcile(ctx, &desired.Spec.GoflowKube); err != nil {
 		log.Error(err, "Failed to get FlowCollector")
 		return ctrl.Result{}, err
 	}
+	// ovs-flows-config map for CNO
+	if err := r.flowsConfig.Reconcile(ctx, desired); err != nil {
+		log.Error(err, "Failed to reconcile ovs-flows-config ConfigMap")
+	}
 
 	return ctrl.Result{}, nil
 }
 
+func (r *FlowCollectorReconciler) checkFinalizerStatus(ctx context.Context, desired *flowsv1alpha1.FlowCollector) error {
```

Comment:
side note: I haven't much looked into finalizers yet but maybe we should also use it to clean up other resources that we're creating?

Reply:
This function is for removing resources that are not managed by Kubebuilder, such as the flows-config configmap. I guess that CRDs that are created/manager by Kubebuilder should be removed automatically. But definitely we should check.